src/reconcile/rec_visibility.c

```diff
@@ -310,12 +311,38 @@ __rec_need_save_upd(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WTI_UPDATE_SELEC
     if (F_ISSET(r, WT_REC_EVICT) && has_newer_updates)
         return (true);
 
-    if (F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED)) {
-        if (upd_select->upd != NULL && !F_ISSET(upd_select->upd, WT_UPDATE_DURABLE))
-            return (true);
+    /*
+     * We need to save the update chain to build the delta. Don't save the update chain if the
+     * selected update is already durable.
+     */
+    if (upd_select->upd != NULL && F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED) &&
+      F_ISSET(&S2C(session)->disaggregated_storage, WT_DISAGG_LEAF_PAGE_DELTA)) {
+        if (upd_select->tombstone != NULL) {
+            if (!F_ISSET(upd_select->tombstone, WT_UPDATE_DURABLE | WT_UPDATE_PREPARE_DURABLE))
+                return (true);
+
+            /* Save the update if we overwrite the previous prepared update. */
+            if (F_ISSET(upd_select->tombstone, WT_UPDATE_PREPARE_DURABLE) &&
+              !upd_select->tw.prepare)
+                return (true);
+        }
 
-        if (upd_select->tombstone != NULL && !F_ISSET(upd_select->tombstone, WT_UPDATE_DURABLE))
-            return (true);
+        if (upd_select->upd->type == WT_UPDATE_TOMBSTONE) {
+            /*
+             * Save the update if we haven't deleted the key from the disk image. We may have
+             * written the tombstone to disk already but we still need to do another delta to remove
+             * it from disk.
+             */
+            if (!F_ISSET(upd_select->upd, WT_UPDATE_DELETE_DURABLE))
```

Comment:
@kommiharibabu We can debate whether we need this. Deleting the key with a stop timestamp in the delta is not helping us saving disk space but actually increases of our disk usage. We need to write a full image to really delete these keys.